pyramid/config/views.py

```diff
@@ -762,6 +776,12 @@ def view(context, request):
         if context is None:
             context = for_
 
+        isexc = isexception(context)
+        if exception_only and not isexc:
+            raise ConfigurationError(
+                'view "context" must be an exception type when '
+                '"exception_only" is True')
```

Comment:
There is nothing that actually requires it to be an exception but it will only be matched for a caught exception so preventing it is a good sanity check.